test/e2e/tuning_options_test.go

```diff
@@ -320,3 +325,219 @@ func TestHAProxyTimeoutsRejection(t *testing.T) {
+			if err := kclient.Get(context.TODO(), icInternalServiceName, icInternalService); err != nil {
+				t.Fatalf("failed to get service %q: %v", icInternalServiceName, err)
+			}
+
+			// Create a cookie that's one character over the max length of a captured cookie. Pad the cookie with a bunch of
+			// 'a's, and include an x at the end that should get truncated. The name of the cookie and the '=' are included in
+			// the max length, so account for that in the size of the padding.
+			padding := strings.Repeat("a", maxLength-len(cookieName)-1)
+			cookieString := fmt.Sprintf("%s=%sx", cookieName, padding)
+
+			curlArgs := []string{
+				"-b", cookieString,
+				// --resolve used this way makes curl send the request to our ingress controller's internal service address as
+				// if it was directed to it by a load balancer. This ensures the request is handled by our ingress controller
+				// without having to wait extra time for a load balancer to be provisioned.
+				"--resolve", fmt.Sprintf("%s:80:%s", echoRoute.Spec.Host, icInternalService.Spec.ClusterIP),
+			}
+
+			curlPod := buildCurlPod(curlPodName.Name, curlPodName.Namespace, curlPodImage, echoRoute.Spec.Host, "", curlArgs...)
+			if err := kclient.Create(context.TODO(), curlPod); err != nil {
+				t.Fatalf("failed to create pod %q: %v", curlPodName.Name, err)
+			}
+			defer assertDeleted(t, kclient, curlPod)
+
+			// Wait for the curl pod to complete its request
+			err = wait.PollImmediate(2*time.Second, 3*time.Minute, func() (bool, error) {
+				if err := kclient.Get(context.TODO(), curlPodName, curlPod); err != nil {
+					t.Logf("failed to get client pod %q: %v", curlPodName, err)
+					return false, nil
+				}
+				return curlPod.Status.Phase == corev1.PodSucceeded, nil
+			})
+
+			if err != nil {
+				t.Fatalf("timed out waiting for pod %q to transition to \"Succeeded\". current phase is %q. error: %v", curlPodName, curlPod.Status.Phase, err)
+			}
+
+			// Find the router pod for our ingress controller so we can check its logs. This ingress controller specifies
+			// replicas=1, so there will only be one.
+			routerPodList, err := getPods(t, kclient, routerDeployment)
+			if err != nil {
+				t.Fatalf("failed to get pods in deployment %s: %v", routerDeploymentName, err)
+			} else if len(routerPodList.Items) != 1 {
+				t.Fatalf("expected to find 1 pod in deployment %s, found %d", routerDeploymentName, len(routerPodList.Items))
+			}
+			routerPod := routerPodList.Items[0]
+
+			// Get the logs from the router pod.
+			kubeConfig, err := config.GetConfig()
+			if err != nil {
+				t.Fatalf("failed to get kube config: %v", err)
+			}
+
+			client, err := kubernetes.NewForConfig(kubeConfig)
+			if err != nil {
+				t.Fatalf("failed to create kube client: %v", err)
+			}
+
+			logsContainerName := "logs"
+			readCloser, err := client.CoreV1().Pods(routerPod.Namespace).GetLogs(routerPod.Name, &corev1.PodLogOptions{
```

Comment:
The pattern that is used for scanning logs usually has a poll loop, like https://github.com/openshift/cluster-ingress-operator/blame/master/test/e2e/http_header_buffer_test.go#L134-L163.  If it isn't there the first time you check, it may not have arrived yet.